frontend/public/components/environment.jsx

```diff
@@ -80,6 +80,149 @@ const stateToProps = ({k8s}, {obj}) => ({
   model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
 });
 
+class CurrentEnvVars {
+  constructor() {
+    this.currentEnvVars = {};
+  }
+
+  instantiateWithRawData(rawEnvData) {
+    this.isContainerArray = _.isArray(rawEnvData.containers);
+    this.hasInitContainers = !_.isUndefined(rawEnvData.initContainers);
+
+    if (this.isContainerArray) {
+      this.currentEnvVars.containers = envVarsToArray(rawEnvData.containers);
+      this.currentEnvVars.initContainers = envVarsToArray(rawEnvData.initContainers);
+    } else {
+      this.currentEnvVars.buildObject = envVarsToArray(rawEnvData);
+    }
+    return this;
+  }
+
+  instantiateWithResultObject(resultObject, isContainerArray, path) {
+    if (isContainerArray) {
+      return this.instantiateWithRawData(_.get(resultObject, _.dropRight(path)));
+    }
+    return this.instantiateWithRawData(_.get(resultObject, path));
+  }
+
+  getEnvVarByTypeAndKey(containerType, key) {
+    if (this.isContainerArray) {
+      return this.currentEnvVars[containerType][key];
+    }
+    return this.currentEnvVars.buildObject[key];
+  }
+
+  setFormattedVars(containerType, index, environmentType, formattedPairs) {
+    if (this.isContainerArray) {
+      this.currentEnvVars[containerType][index][environmentType] = formattedPairs;
+    } else {
+      this.currentEnvVars.buildObject[index][environmentType] = formattedPairs;
+    }
+    return this;
+  }
+
+  /**
+   * Return true if env vars are equal.
+   *
+   *
+   * @param testEnvData
+   * @returns boolean
+   * @public
+   */
+  isEqual(testEnvData) {
+    if (this.isContainerArray) {
+      return _.isEqual(testEnvData.currentEnvVars.containers, this.currentEnvVars.containers) && _.isEqual(testEnvData.currentEnvVars.initContainers, this.currentEnvVars.initContainers);
+    }
+    return _.isEqual(testEnvData.currentEnvVars.buildObject, this.currentEnvVars.buildObject);
```

Comment:
Is there a reason this can't just be `_.isEqual(currentEnvVars, testEnvData)`? Are there other properties we're purposely not checking?